iroh/src/node.rs

```diff
@@ -586,11 +606,13 @@ impl RpcHandler {
     ) -> impl Stream<Item = ValidateProgress> + Send + 'static {
         let (tx, rx) = mpsc::channel(1);
         let tx2 = tx.clone();
-        self.rt().main().spawn(async move {
-            if let Err(e) = self.inner.db.validate(tx).await {
-                tx2.send(ValidateProgress::Abort(e.into())).await.unwrap();
-            }
-        });
+        if let Some(db) = self.concrete_db() {
+            self.rt().main().spawn(async move {
+                if let Err(e) = db.validate(tx).await {
```

Comment:
`validate` seems potentially very useful for different types of backends, not just the `Database`?

Reply:
Yeah, I think this will have to go into another trait or be implemented in terms of stuff that the database trait provides. I guess you would only need a way to list all blobs. But it would be a major refactor because the current validate uses sync IO and the new generic IO is by default async. I guess this is why I did this hack...

Reply:
Added another trait that has validate